provisio-core/src/main/java/ca/vanzyl/provisio/action/artifact/WriteToDiskAction.java

```diff
@@ -43,7 +43,8 @@ public WriteToDiskAction(ProvisioArtifact artifact, File outputDirectory) {
     public void execute(ProvisioningContext context) {
         File file = artifact.getFile();
         if (file != null) {
-            String targetName = artifact.getName() != null ? artifact.getName() : file.getName();
+            String targetName =
+                    artifact.getGroupId() + '_' + (artifact.getName() != null ? artifact.getName() : file.getName());
```

Comment:
is `artifact.getGroupId()` always non-null?

also, can we have detection that files we create are unique?
ie.in the problem described in https://github.com/trinodb/trino/issues/23187, i would expect the build to fail because it wanted to put two (different) jars at the same path. clearly some check could be added

Reply:
we can fail the build simply by removing REPLACE_EXISTING during copy.

Reply:
But I'm not sure whether this will break build if the provisio is executed without a clean. I'll leave it as it is for now.